core/blockchain.go

```diff
@@ -2018,6 +2029,105 @@ func (bc *BlockChain) update() {
 	}
 }
 
+// updateTxIndices is responsible for the construction and deletion of the
+// transaction index.
+//
+// User can use flag `txlookuplimit` to specify a "recentness" block, below
+// which ancient tx indices get deleted. If `txlookuplimit` is 0, it means
+// all tx indices will be reserved.
+//
+// The user can adjust the txlookuplimit value for each launch, Geth will
+// automatically construct the missing indices and delete the extra indices.
+func (bc *BlockChain) updateTxIndices() {
+	var (
+		done   chan struct{} // Non-nil if background unindexing or reindexing routine is active.
+		headCh = make(chan ChainHeadEvent)
+	)
+	sub := bc.SubscribeChainHeadEvent(headCh)
+	defer func() {
+		if sub != nil {
+			sub.Unsubscribe()
+		}
+	}()
```

Comment:
Lets move the subscription creation and defer below the method declarations. Seems cleaner code that we subscribe and immediately afterwards have the handler logic.